src/cc/api/BPF.cc

```diff
@@ -671,7 +671,7 @@ int BPF::poll_perf_buffer(const std::string& name, int timeout_ms) {
 }
 
 StatusTuple BPF::load_func(const std::string& func_name, bpf_prog_type type,
-                           int& fd, unsigned flags) {
+                           int& fd, unsigned flags, bpf_attach_type attach_type) {
```

Comment:
It would be good to change the argument name to expected_attach_type. This is to load_func and not really do any attachment. The expected_attach_type is also consistent with the field name in the attribute.